<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PartGui::DlgSettingsGeneral</class>
 <widget class="QWidget" name="PartGui::DlgSettingsGeneral">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>552</width>
    <height>486</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>General</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Model settings</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0" colspan="2">
       <widget class="Gui::PrefCheckBox" name="checkBooleanCheck">
        <property name="text">
         <string>Automatically check model after boolean operation</string>
        </property>
        <property name="prefEntry" stdset="0">
         <cstring>CheckModel</cstring>
        </property>
        <property name="prefPath" stdset="0">
         <cstring>Mod/Part/Boolean</cstring>
        </property>
       </widget>
      </item>
      <item row="1" column="0" colspan="2">
       <widget class="Gui::PrefCheckBox" name="checkBooleanRefine">
        <property name="text">
         <string>Automatically refine model shape of features in Part workbench</string>
        </property>
        <property name="prefEntry" stdset="0">
         <cstring>RefineModel</cstring>
        </property>
        <property name="prefPath" stdset="0">
         <cstring>Mod/Part/Boolean</cstring>
        </property>
       </widget>
      </item>
      <item row="2" column="0" colspan="2">
       <widget class="Gui::PrefCheckBox" name="checkSketchBaseRefine">
        <property name="text">
         <string>Automatically refine model shape of features in PartDesign workbench</string>
        </property>
        <property name="prefEntry" stdset="0">
         <cstring>RefineModel</cstring>
        </property>
        <property name="prefPath" stdset="0">
         <cstring>Mod/PartDesign</cstring>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="Gui::PrefCheckBox" name="checkLinearize">
        <property name="toolTip">
         <string>Try to replace flat BSpline face with simple planar face.
Currently only applies to Loft, Sweep, drafted Extrusion
in Part workbench, and equivalent features in PartDesign.</string>
        </property>
        <property name="text">
         <string>Linearize flat BSpline face when possible</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
        <property name="prefPath" stdset="0">
         <cstring>Mod/Part</cstring>
        </property>
        <property name="prefEntry" stdset="0">
         <cstring>LinearizeExtrusionDraft</cstring>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="Gui::PrefCheckBox" name="checkSplitEllipsoid">
        <property name="text">
         <string>Split ellipsoid</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
        <property name="prefPath" stdset="0">
         <cstring>Mod/Part</cstring>
        </property>
        <property name="prefEntry" stdset="0">
         <cstring>SplitEllipsoid</cstring>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="Gui::PrefCheckBox" name="checkSingleSolid">
        <property name="text">
         <string>Enforce single solid in Body</string>
        </property>
        <property name="prefEntry" stdset="0">
         <cstring>SingleSolid</cstring>
        </property>
        <property name="prefPath" stdset="0">
         <cstring>Mod/Part</cstring>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="Gui::PrefCheckBox" name="checkAutoGroupSolids">
        <property name="text">
         <string>Auto group solids in Body</string>
        </property>
        <property name="prefPath" stdset="0">
         <cstring>Mod/Part</cstring>
        </property>
        <property name="prefEntry" stdset="0">
         <cstring>AutoGroupSolids</cstring>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="Gui::PrefCheckBox" name="checkBoxValidateShape">
        <property name="toolTip">
         <string>Validate on changing shape. If the shape is invalid, a warning icon
will appear in the corresponding tree view item. Be aware that enabling
this option on complex shape may cause slow down.

Note, this option only applies to newly created object. For existing
object, you can manually change its 'ValidShape' property.</string>
        </property>
        <property name="text">
         <string>Validate shape</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
        <property name="prefEntry" stdset="0">
         <cstring>AutoValidateShape</cstring>
        </property>
        <property name="prefPath" stdset="0">
         <cstring>Mod/Part</cstring>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="Gui::PrefCheckBox" name="checkBoxFixShape">
        <property name="toolTip">
         <string>Try to fix invalid shape on changing. Note, this option only applies to
newly created object.</string>
        </property>
        <property name="text">
         <string>Fix invalid shape</string>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
        <property name="prefEntry" stdset="0">
         <cstring>FixShape</cstring>
        </property>
        <property name="prefPath" stdset="0">
         <cstring>Mod/Part</cstring>
        </property>
       </widget>
      </item>
      <item row="6" column="0" colspan="2">
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="Gui::PrefCheckBox" name="checkBoxAuxGroup">
          <property name="toolTip">
           <string>Disable this option to not enforce unique label in auxiliary groups.
So instead of having 'Datums001' etc., all datum group can be
labeled as 'Datums'.

Note that this option does not apply to user created auxiliary
groups, neither does the option affects any user defined label of
existing auxiliary groups.</string>
          </property>
          <property name="text">
           <string>Use unique label for auxiliary groups in Body</string>
          </property>
          <property name="prefEntry" stdset="0">
           <cstring>AuxGroupUniqueLabel</cstring>
          </property>
          <property name="prefPath" stdset="0">
           <cstring>Mod/Part</cstring>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btnAuxGroup">
          <property name="toolTip">
           <string>Apply the 'fixed name' option to existing auxiliary groups.
it only applies to auto created group with their original labels.</string>
          </property>
          <property name="text">
           <string>Apply to existing groups</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="7" column="0">
       <widget class="Gui::PrefCheckBox" name="checkAutoAuxiliaryGrouping">
        <property name="text">
         <string>Automatic auxiliary grouping for new Bodies</string>
        </property>
        <property name="prefPath" stdset="0">
         <cstring>Mod/Part</cstring>
        </property>
        <property name="prefEntry" stdset="0">
         <cstring>AutoAuxGrouping</cstring>
        </property>
       </widget>
      </item>
      <item row="7" column="1">
       <widget class="Gui::PrefCheckBox" name="checkAutoHideOrigins">
        <property name="text">
         <string>Automatically hide Origins in Tree view</string>
        </property>
        <property name="prefPath" stdset="0">
         <cstring>Mod/Part</cstring>
        </property>
        <property name="prefEntry" stdset="0">
         <cstring>AutoHideOrigins</cstring>
        </property>
       </widget>
      </item>
      <item row="8" column="0">
       <widget class="Gui::PrefCheckBox" name="checkEnforceOpsPrecision">
        <property name="toolTip">
         <string>Automatically round operations which have failed. For example, a chamfer which is set to exactly the length of an edge may fail due to being slightly over the edge length, so rounding down to just below the edge length will lead to a successful chamfer.</string>
        </property>
        <property name="text">
         <string>Enforce Numeric Precision</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
        <property name="prefPath" stdset="0">
         <cstring>Mod/Part</cstring>
        </property>
        <property name="prefEntry" stdset="0">
         <cstring>EnforcePrecision</cstring>
        </property>
       </widget>
      </item>
      <item row="8" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QLabel" name="label_3">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="baseSize">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
          <property name="text">
           <string>Precision Level (Decimal Places)</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="Gui::PrefSpinBox" name="spinBoxPrecisionLevel">
          <property name="toolTip">
           <string>Number of decimal places (preicison) to use internall for part designer operations.</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
          <property name="suffix">
           <string/>
          </property>
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>34</number>
          </property>
          <property name="value">
           <number>12</number>
          </property>
          <property name="prefEntry" stdset="0">
           <cstring>OpsPrecisionLevel</cstring>
          </property>
          <property name="prefPath" stdset="0">
           <cstring>Mod/Part</cstring>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_3">
     <property name="visible">
      <bool>true</bool>
     </property>
     <property name="title">
      <string>Object naming</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0">
       <widget class="Gui::PrefCheckBox" name="checkObjectNaming">
        <property name="text">
         <string>Add name of base object</string>
        </property>
        <property name="prefPath" stdset="0">
         <cstring>Mod/Part</cstring>
        </property>
        <property name="prefEntry" stdset="0">
         <cstring>UseBaseObjectName</cstring>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Design behavior</string>
     </property>
     <layout class="QFormLayout" name="formLayout_3">
      <property name="fieldGrowthPolicy">
       <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Override commands</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="Gui::PrefComboBox" name="comboBoxCommandOverride">
        <property name="toolTip">
         <string>Override non-PartDesign command with an equivalent PartDesign
command when operate on PartDesign feature.</string>
        </property>
        <property name="currentIndex">
         <number>2</number>
        </property>
        <property name="prefPath" stdset="0">
         <cstring>Mod/Part</cstring>
        </property>
        <property name="prefEntry" stdset="0">
         <cstring>CommandOverride</cstring>
        </property>
        <item>
         <property name="text">
          <string>Disable</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Enable</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Always ask</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Auto wrap feature</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="Gui::PrefComboBox" name="comboBoxWrapFeature">
        <property name="toolTip">
         <string>If a non-PartDesign object references a PartDesign feature, incorporate
the object into PartDesign body using a wrap feature.</string>
        </property>
        <property name="currentIndex">
         <number>2</number>
        </property>
        <property name="prefPath" stdset="0">
         <cstring>Mod/Part</cstring>
        </property>
        <property name="prefEntry" stdset="0">
         <cstring>EnableWrapFeature</cstring>
        </property>
        <item>
         <property name="text">
          <string>Disable</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Enable</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Always ask</string>
         </property>
        </item>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer>
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Gui::PrefCheckBox</class>
   <extends>QCheckBox</extends>
   <header>Gui/PrefWidgets.h</header>
  </customwidget>
  <customwidget>
   <class>Gui::PrefComboBox</class>
   <extends>QComboBox</extends>
   <header>Gui/PrefWidgets.h</header>
  </customwidget>
  <customwidget>
   <class>Gui::PrefSpinBox</class>
   <extends>QSpinBox</extends>
   <header>Gui/PrefWidgets.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
